diesel/src/macros/mod.rs

```diff
@@ -211,8 +213,8 @@ macro_rules! joinable_inner {
 macro_rules! allow_tables_to_appear_in_same_query {
     ($left_mod:ident, $($right_mod:ident),+ $(,)*) => {
         $(
-            impl $crate::query_source::TableNotEqual<$left_mod::table> for $right_mod::table {}
-            impl $crate::query_source::TableNotEqual<$right_mod::table> for $left_mod::table {}
+            impl $crate::query_source::ViewNotEqual<$left_mod::table> for $right_mod::table {}
```

Comment:
Hmm in some other places we name this around `Source` as the common denominator of view and table, it looks like this may be something we want to be consistent with here.